internal/rukpak/util/util_test.go

```diff
@@ -0,0 +1,148 @@
+package util_test
+
+import (
+	"errors"
+	"io"
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/operator-framework/operator-controller/internal/rukpak/util"
+)
+
+func TestMergeMaps(t *testing.T) {
+	tests := []struct {
+		name      string
+		maps      []map[string]string
+		expectMap map[string]string
+	}{
+		{
+			name:      "no maps",
+			maps:      make([]map[string]string, 0),
+			expectMap: map[string]string{},
+		},
+		{
+			name:      "two empty maps",
+			maps:      []map[string]string{{}, {}},
+			expectMap: map[string]string{},
+		},
+		{
+			name:      "simple add",
+			maps:      []map[string]string{{"foo": "bar"}, {"bar": "foo"}},
+			expectMap: map[string]string{"foo": "bar", "bar": "foo"},
+		},
+		{
+			name:      "subsequent maps overwrite prior",
+			maps:      []map[string]string{{"foo": "bar", "bar": "foo"}, {"foo": "foo"}, {"bar": "bar"}},
+			expectMap: map[string]string{"foo": "foo", "bar": "bar"},
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			assert.Equalf(t, tc.expectMap, util.MergeMaps(tc.maps...), "maps did not merge as expected")
+		})
+	}
+}
+
+// Mock reader for testing that always returns an error when Read is called
+type errorReader struct {
+	io.Reader
+}
+
+func (m errorReader) Read(p []byte) (int, error) {
+	return 0, errors.New("Oh no!")
+}
+
+func TestManifestObjects(t *testing.T) {
+	tests := []struct {
+		name          string
+		wantErr       bool
+		yaml          string
+		expectObjects []client.Object
+		reader        func(string) io.Reader
+	}{
+		{
+			name:          "nothing",
+			wantErr:       false,
+			yaml:          "",
+			expectObjects: make([]client.Object, 0),
+		},
+		{
+			name:    "happy path",
+			wantErr: false,
+			yaml: `apiVersion: v1
+kind: Service
+metadata:
+  name: service-a
+  namespace: ns-a
+spec:
+  clusterIP: None
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: service-b
+  namespace: ns-b
+spec:
+  clusterIP: 0.0.0.0`,
```

Comment:
wrong formatting?

Reply:
I'm defining a file inline here literally with the tilde character instead of creating a separate file for it. It'll look odd compared to the formatting of the rest of the code here because it doesn't have leading whitespace unless required for yaml format - assuming that's what you're referring to, or did you see an issue in the yaml content?

Reply:
Got it. All good. Thanks